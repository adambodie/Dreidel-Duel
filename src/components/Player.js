import React, { Component } from "react";
import styles from "./player.module.css";


export default class Player extends Component {
  constructor(props) {
    super(props);
    this.state = {
		value: ''
	}
      
	this.handleChange = this.handleChange.bind(this);
	this.handleNameChange = this.handleNameChange.bind(this);
  }
	handleChange(e) {
		this.props.onScoreChange(e.target.value);  
  }
	handleNameChange(e) {
		this.setState({value: e.target.value});
	}

  render() {
    return (
		<div className={styles.player}>
			<h2>Player {this.props.id}</h2>
			<input type="text" value={this.state.value} onChange={this.handleNameChange} />
			<div className={styles.counter}>
				<button
					className={styles.counterAction}
					onClick={this.handleChange}
					disabled={this.props.disabled}
					>
					SPIN
				</button>
				<div className={styles.counterScore}>
					{this.props.score}
				</div>
			</div>
		</div>
				
    );
  }
}
