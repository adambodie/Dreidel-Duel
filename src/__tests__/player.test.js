import React from "react"
import renderer from "react-test-renderer"
import Player from "../components/player"

describe("Player", () =>
  it("renders correctly", () => {
    const tree = renderer.create(<Player />).toJSON()
    expect(tree).toMatchSnapshot()
  }))

