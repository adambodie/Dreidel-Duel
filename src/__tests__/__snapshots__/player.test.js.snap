// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders correctly 1`] = `
<div
  className="player"
>
  <h2>
    Player 
  </h2>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="counter"
  >
    <button
      className="counterAction"
      onClick={[Function]}
    >
      SPIN
    </button>
    <div
      className="counterScore"
    />
  </div>
</div>
`;
